src/horust/supervisor/process_spawner.rs

```diff
@@ -173,6 +202,43 @@ fn redirect_output(
     Ok(())
 }
 
+fn open_next_chunk(base_path: &Path) -> io::Result<File> {
+    let mut count = 1;
+    let mut path = base_path.to_path_buf();
+
+    while path.is_file() {
+        // let path_str = format!("{}.{count}", base_path,  path.to_string_lossy().to_string() + format!(".{count}").as_ref();
+        let path_str = format!("{}.{count}", base_path.to_string_lossy());
+        // path = Path::new(&path_str).to_path_buf();
+        path = Path::new(&path_str).to_path_buf();
+        count += 1;
+    }
+    debug!("Opening next output: {}", path.display());
+    OpenOptions::new().create(true).append(true).open(&path)
+}
+
+fn chunked_writer(fd: OwnedFd, path: &Path, file_size: u32) -> Result<()> {
+    let path = path.to_path_buf();
+
+    std::thread::spawn(move || {
+        let error_printer = |e| error!("Error: {e}");
+        let source = File::from(fd);
+        loop {
+            let mut reader = BufReader::new(&source).take(file_size as u64);
```

Comment:
reader can be outside the loop I think?

Reply:
The logic is that there is one main file descriptor outside of the loop, which is then wrapped by a BufReader and a `take()` with a byte limit inside the loop.
This allows the loop to create files of specified sizes by EOF'ing on the inner BufReader.